internal/resources/cluster/data_source_cluster.go

```diff
@@ -131,11 +131,17 @@ func dataSourceTMCClusterRead(ctx context.Context, d *schema.ResourceData, m int
 
 	switch {
```

Comment:
Just a thought, instead of doing this manually, @Vasundharashukla can we use the marshal and unmarshal binary methods available for the defined spec model? In this way, we may be able to avoid the checks for different types of clusters. cc: @shreyassreenivas @shobha2626

Reply:
Here the use case and the context is different